src/PipeSpec.php

```diff
@@ -0,0 +1,42 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Laminas\Mvc\Middleware;
+
+use Closure;
+use Psr\Http\Server\MiddlewareInterface;
+use Psr\Http\Server\RequestHandlerInterface;
+use Webmozart\Assert\Assert;
+
+final class PipeSpec
+{
+    /**
+     * @var array
```

Comment:
`@var list<string|RequestHandlerInterface|MiddlewareInterface|Closure>`

Reply:
@rieschl That would have to be `@psalm-var` to work correctly with our phpcs rules, I think. But yes, adding it now just helps set the expectations and make it more explicit.